app/src/organisms/FirmwareUpdateModal/FirmwareUpdateTakeover.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react'
+import {
+  useInstrumentsQuery,
+  useCurrentMaintenanceRun,
+} from '@opentrons/react-api-client'
+import { UpdateNeededModal } from './UpdateNeededModal'
+
+export function FirmwareUpdateTakeover(): JSX.Element {
+  const [
+    showUpdateNeededModal,
+    setShowUpdateNeededModal,
+  ] = React.useState<boolean>(false)
+  const instrumentsData = useInstrumentsQuery({
+    refetchInterval: 5000,
+  }).data?.data
+  const { data: maintenanceRunData } = useCurrentMaintenanceRun()
+  const subsystemUpdateInstrument = instrumentsData?.find(
+    instrument => instrument.ok === false
+  )
+
+  React.useEffect(() => {
+    if (subsystemUpdateInstrument != null && maintenanceRunData == null) {
+      setShowUpdateNeededModal(true)
+    }
+  }, [subsystemUpdateInstrument])
+
+  return (
+    <>
+      {subsystemUpdateInstrument != null && showUpdateNeededModal && (
+        <UpdateNeededModal
+          subsystem={subsystemUpdateInstrument.subsystem}
+          setShowUpdateModal={setShowUpdateNeededModal}
+        />
+      )}
```

Comment:
mostly recent code uses `? <something > : null for displaying modal`

```suggestion
      {subsystemUpdateInstrument != null && showUpdateNeededModal ? (
        <UpdateNeededModal
          subsystem={subsystemUpdateInstrument.subsystem}
          setShowUpdateModal={setShowUpdateNeededModal}
        />
      ) : null }
```